src/runners/junitRunner/JUnitRunnerResultAnalyzer.ts

```diff
@@ -29,13 +29,20 @@ export class JUnitRunnerResultAnalyzer extends BaseRunnerResultAnalyzer {
                 return;
             }
 
-            if (!this.testResults.has(testFullName)) {
+            let detail: ITestResultDetails | undefined = this.testResults.get(testFullName);
+            if (!detail) {
                 this.currentTestItem = testFullName;
-                const detail: ITestResultDetails = { status: undefined };
+                detail = { status: undefined };
                 if (data.indexOf(MessageId.IGNORE_TEST_PREFIX) > -1) {
                     detail.status = TestStatus.Skip;
+                } else {
+                    detail.duration = - Date.now();
                 }
                 this.testResults.set(testFullName, detail);
+            } else if (detail.duration !== undefined) {
+                // Some test cases may executed multiple times (@RepeatedTest), we need to calculate the time for each execution,
+                // so here plus is used.
+                detail.duration += - Date.now();
```

Comment:
hard to understand it.  `duration -= startTime;`

Reply:
Updated